modules/tinymce/src/core/main/ts/api/WindowManager.ts

```diff
@@ -78,20 +82,27 @@ const WindowManager = (editor: Editor): WindowManager => {
     });
   };
 
-  const addDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>) => {
-    dialogs.push(dialog);
+  const addDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>, triggerElement: Optional<SugarElement<HTMLElement>>) => {
+    dialogs.push({ instanceApi: dialog, triggerElement });
     fireOpenEvent(dialog);
   };
 
   const closeDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>) => {
     fireCloseEvent(dialog);
-    dialogs = Arr.filter(dialogs, (otherDialog) => {
-      return otherDialog !== dialog;
+
+    Arr.findIndex(dialogs, ({ instanceApi }) => instanceApi === dialog).each( (dialogIndex) => {
+      const dialogTriggerElement = dialogs[dialogIndex].triggerElement;
+
+      dialogs.splice(dialogIndex, 1);
```

Comment:
I'm not a huge fan of `splice`, and I also don't like that if the instance isn't found focus will never be set anywhere.

- [ ] switch back to `filter` and use a `let` variable to store the trigger element inside the filter check